alumnos/nelb/ejercicios/clase-3.js

```diff
@@ -0,0 +1,208 @@
+// 2PIr
+function perimetroCirculo(r) {
+  return 2*Math.PI*r;
+}
+
+console.assert(perimetroCirculo(3) === 18.84955592153876)
+console.assert(perimetroCirculo(6) === 37.69911184307752)
+// 1
+function areaCirculo(r) {
+  return Math.PI*r**2;
+}
+
+console.assert(areaCirculo(2) === 12.566370614359172)
+console.assert(areaCirculo(8) === 201.06192982974676)
+
+// 2
+
+function hipotenusa(c1, c2) {
+  return Math.sqrt(c1**2 + c2**2);
+}
+console.assert(hipotenusa(1, 2) === 2.23606797749979)
+
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min)) + min;
+}
+
+function azarMoneda() {
+  return Math.random() > 0.5 ? 'Cara' : 'cruz';
+}
+
+function azarMoneda() {
+  return getRandomInt(0,1) === 0 ? 'Cara' : 'cruz';
+}
+
+azarMoneda();
+
+// 3
+
+function azarDados() {
+  let numVeces = 0;
+  for (let i =0; i < 100; i++){
+    if (getRandomInt(1,6) + getRandomInt(1,6) === 10){
+      numVeces++;
+    }
+  }
+  return numVeces
+}
+
+azarDados();
+
+// 4
+
+function repeat(str, n) {
+    return str.repeat(n);
+
+}
+
+console.assert(repeat('Batman ', 3) === 'Batman Batman Batman ')
+
+// 5
+
+function palabraMasLarga(str) {
+  const palabras =  str.split(' ');
+  let palabra = palabras[0];
+  for (let i = 0; i < palabras.length -1; i++) {
+    palabra = palabra.length > palabras[i + 1].length ? palabra : palabras[i + 1].toString();
```

Comment:
por qué en caso de que la palabra que estás evaluando no sea la más larga te quedas con la siguiente? 😅

Reply:
Porque son muy malas horas de curso jejejeje. Lo que pasa es que el for lo empiezo en 0 y entonces al asignar a palabra el primer valor del array empiezo a compararla siempre con el valor de al siguiente a i. Vamos, que he debido empezar el for en 1 y evaluar con palabras[i] pero no lo vi en ese momento.